src/core/api/index.js

```diff
@@ -265,3 +266,42 @@ export function autocomplete({ api, filters }: AutocompleteParams) {
     state: api,
   });
 }
+
+type GetNextResponseType =
+  (nextURL?: string) => Promise<PaginatedApiResponse<any>>;
+
+type AllPagesOptions = {| pageLimit: number |};
+
+export const allPages = async (
```

Comment:
This is neat and probably fine for now but as mentioned it might be nice to request an API for this. I'd think a single request/endpoint would be better than what we're doing here 😄

Reply:
I removed the code that makes a lot of API requests but we still need to fetch all of the user's collections to fill up the select box.